101-webapp-linux-sonarqube-mysql/azuredeploy.json

```diff
@@ -0,0 +1,218 @@
+    "hostingPlanName": "[concat(parameters('siteName'), 'serviceplan')]"
+
+  },
+  "resources": [
+    {
+      "comments":"This is the Linux web app with the Sonarquebe Docker image and the DB",
+      "type": "Microsoft.Web/sites",
+      "name": "[parameters('siteName')]",
+      "properties": {
+        "siteConfig": {
+          "linuxFxVersion": "DOCKER|SONARQUBE"
+        },
+        "name": "[parameters('siteName')]",
+        "serverFarmId": "[variables('hostingPlanName')]",
+        "hostingEnvironment": ""
+      },
+      "apiVersion": "2016-03-01",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[variables('hostingPlanName')]",
+        "[variables('databaseName')]"
+      ],
+      "resources": [
+        {
+          "name": "appsettings",
+          "type": "config",
+          "apiVersion": "2015-08-01",
+          "comments": "This are the generated settings for the Sonar jdbc connectionstring",
+          "dependsOn": [
+            "[resourceId('Microsoft.Web/sites', parameters('siteName'))]"
+          ],
+          "tags": {
+            "displayName": "SonarappSettings"
+          },
+          "properties": {
+            "SONARQUBE_JDBC_URL": "[concat('jdbc:mysql://', variables('serverName') ,'.mysql.database.azure.com:3306/',variables('databaseName'),'?verifyServerCertificate=true&useSSL=true&requireSSL=false&useUnicode=true&characterEncoding=utf8')]",
+            "SONARQUBE_JDBC_USERNAME": "[variables('jdbcSonarUserName')]",
+            "SONARQUBE_JDBC_PASSWORD": "[parameters('administratorLoginPassword')]"
+          }
+        }
+      ]
+    },
+    {
+      "apiVersion": "2016-09-01",
+      "name": "[variables('hostingPlanName')]",
+      "type": "Microsoft.Web/serverfarms",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "name": "[variables('hostingPlanName')]",
+        "workerSizeId": "1",
+        "reserved": true,
+        "numberOfWorkers": "1",
+        "hostingEnvironment": ""
+      },
+      "sku": {
+        "Tier": "[parameters('servicePlanTier')]",
+        "Name": "[parameters('servicePlanSku')]"
+      },
+      "kind": "linux"
+    },
+    {
+      "apiVersion": "2016-02-01-privatepreview",
+      "kind": "",
+      "location": "[resourceGroup().location]",
+      "name": "[variables('serverName')]",
+      "properties": {
+        "version": "[parameters('mysqlVersion')]",
+        "administratorLogin": "[parameters('administratorLogin')]",
+        "administratorLoginPassword": "[parameters('administratorLoginPassword')]",
+        "storageMB": "[parameters('databaseSkuSizeMB')]"
+      },
+      "sku": {
+        "name": "[parameters('databaseSkuName')]",
+        "tier": "[parameters('databaseSkuTier')]",
+        "capacity": "[parameters('databaseDTU')]",
+        "size": "[parameters('databaseSkuSizeMB')]",
+        "family": "SkuFamily"
+      },
+      "type": "Microsoft.DBforMySQL/servers",
```

Comment:
move this to the top of the resource object

Reply:
I suppose you refer to type. Solved in next PR